tests/scripts/sort_configs.py

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env python
+"""Script to sort conf/logs.json schema file"""
+import json
+import os
+import logging
+
+from collections import OrderedDict
+
+SCRIPT_NAME = 'JsonFileSorter'
+CONF_LOGS_FILE = os.path.join(os.path.dirname(__file__), '../../conf/logs.json')
+
+
+class JsonFileSorter(object):
+    """
+    Tests that the conf.json file is formatted properly and is sorted alphabetically
+    on the top-level key
+    """
+
+    def __init__(self):
+        self._logger = logging.getLogger(SCRIPT_NAME)
+
+    def sort_json_file(self, file_path):
+        self._logger.info('Sorting file: {}...'.format(file_path))
+
+        with open(file_path, 'r') as infile:
+            original_text = infile.read().strip()
+
+        # Load the JSON document using OrderedDict so that it preserves original ordering
+        schema = json.loads(original_text, object_pairs_hook=OrderedDict)
+
+        ordered_schema = OrderedDict(sorted(schema.items(), key=lambda k: k[0]))
```

Comment:
we need to be careful w/ csv-type log schemas. the reason OrderedDict is used in the first place here is that csv keys must retain their order. will your script account for keeping key order (regardless of canonical sort order) when `parser` == `csv`?

Reply:
I do notice the csv ones were not reordered in your logs.json, but don't see anything that accounts for it.. 🤔

Reply:
I actually ran into this exact problem. Using `json.loads(....)` without the `OrderDict` hook results a bunch of nested `dicts` to be returned. Even if I wrap it with an `OrderedDict` later, the ordering is already destroyed on-load, so after "sorting" I noticed in the output that all of the internal keys were jumbled up.

I found using the `object_pairs_hook` will return a bunch of nested `OrderedDicts` with the original ordering of the internal keys preserved. The sorting is only concerned with sorting the top-level keys. To my understanding this *should* preserve original ordering of internal keys. 

I'll update the comment to be clearer—I didn't know about the CSV key ordering and I think that's an important detail..